### PR TITLE
fix(node): support `tty.hasColors()` and `tty.getColorDepth()`

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This PR adds support for [`tty.WriteStream.prototype.hasColors()`](https://nodejs.org/api/tty.html#writestreamhascolorscount-env) and [`tty.WriteStream.prototype.getColorDepth()`](https://nodejs.org/api/tty.html#writestreamgetcolordepthenv).

I couldn't find any usage on GitHub which passes parameters to it. Therefore I've skipped adding support for the `env` parameter to keep our snapshot size small.

Based on https://github.com/denoland/deno_terminal/pull/3

Fixes https://github.com/denoland/deno/issues/24616

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]

File: ext/node/polyfills/tty.js [hunk 1]
```diff
@@ -105,6 +106,25 @@ export class WriteStream extends Socket {
...
+    if (env === undefined && typeof count === "object") {
+      env = count;
+      count = 16;
+    }
+
+    // TODO(@marvinhagemeister): Ignore env parameter.
+    // Haven't seen it used anywhere, seems more done
+    // to make testing easier in Node.
+
+    const depth = op_bootstrap_color_depth();
```
Comment (dsherret) [inline ext/node/polyfills/tty.js:125]: Should this be checking this for the current fd rather than using a global value?

Comment (dsherret) [reply ext/node/polyfills/tty.js:125]: It looks like the answer is no to my question.

**Review by dsherret — APPROVED** [review]
LGTM

**Comment by marvinhagemeister** [comment]
Noticed that node also has a `tty.getColorDepth()` method, so I've added that too.

**Pull request merged**